docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,132 @@
+
+- **`{javaSrc}/data/SharedPreferences`**
+  Used to read/write shared preferences for the example app, including the current theme.
+
+
+## Backpack
+
+The backpack lib is where all components live. All packages, except `utils` are Backpack components.
+
+### Component's architecture
+
+There are only a few things that should be observed about the architecture of the components: 
+
+1. Whenever possible the Backpack component should extend from the Android base component and add the additional extra functionality instead of wrapping the Android component inside another `ViewGroup`. This is to keep the view hierarchy as small as possible.
+2. All props should be available via XML and programmatically. Whenever a component is available in both XML and code all its properties should also be available in both. 
+3. Changing a property programmatically should immediately reflect on the UI.
+4. Components should be "extensible", i.e. `open class` in Kotlin. It's also important to think about the component being used that way when writing/reviewing it.
+
+More can be found here: https://github.com/Skyscanner/backpack-android/tree/master/decisions
+
+### Theming
+- Uses default Android theming mechanism https://developer.android.com/guide/topics/ui/look-and-feel/themes
+- Each themeable component exposes a global attribute that can be used to theme all instances of that component.
+  - E.g. BpkChip can be themed in two ways:
+    ```xml
+    // app/res/values/themes.xml
+
+    <style name="RedChip">
+      <item name="chipSelectedBackgroundColor">@colors/bpkRed500</item>
+    </style>
+    ```
+    1. Individually through the `style` prop:
+    ```xml
+      <BpkChip style="@style/RedChip" />
+    ```
+    2. Globally via the `bpkChipStyle` attribute:
+    ```xml
+    // app/res/values/style.xml
+    <style name="RedChip">
+      <item name="chipSelectedBackgroundColor">@colors/bpkRed500</item>
+    </style>
+
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
+      <item name="bpkChipStyle">@style/RedChip</item>
+    </style>
+    ```
+- Themes can also be set programmatically using a `ContextThemeWrapper`:
+  ```Kotlin
+  val themedContext = ContextThemeWrapper(context, R.style.RedChip)
+  val chip = BpkChip(themedContext)
+  ```
+
+### Caveats
+
+The `style` attribute will take precedence over the global `bpkChipStyle`, meaning it is possible to
+have a different style for a single component.
+
+It is also possible to set theming properties directly via `xml`:
+```xml
+  <BpkChip style="@style/RedChip" app:chipSelectedBackgroundColor="@colors/bpkRed500" />
+```
+This is not advertised and is not how we intend theming to be used.
+
+
+## Testing
+
+There are currently three types of tests in `backpack-android`:
+
+- Connected tests:
+  - Live inside the `Backpack` module and run inside an emulator, hence "connected". This includes mostly unit testing for visual components.
```

Comment:
```suggestion
  - Live inside the `backpack` module and run inside an emulator, hence "connected". This includes mostly unit testing for visual components.
```

Not sure which of capitalised or lower case is better for the module name but it's done in two ways in this file, so my suggestions change it to lower case everywhere. If that isn't right though go for capitalised 😄

Reply:
well the actual gradle module name is upper case. I'll review it and make them upper case